### PR TITLE
CAMEL-20156: camel-openapi-java - attempt to convert allowValues by S…

### PR DESCRIPTION
**Pull request by zhfeng**
…chema::cast method

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: components/camel-openapi-java/src/test/java/org/apache/camel/openapi/RestOpenApiReaderTest.java [hunk 1]
```diff
@@ -170,6 +251,14 @@ public void testReaderRead() throws Exception {
         assertTrue(json.contains("\"success\" : \"123\""));
         assertTrue(json.contains("\"error\" : \"-1\""));
         assertTrue(json.contains("\"type\" : \"array\""));
+        assertTrue(json.contains("\"enum\" : [ \"A\", \"B\", \"C\" ]"));
+        assertTrue(json.contains("\"enum\" : [ 1, 2, 3 ]"));
+        assertTrue(json.contains("\"enum\" : [ 1.0, 2.0, 3.0 ]"));
+        assertTrue(json.contains("\"enum\" : [ \"true\", \"false\" ]"));
```
Comment (zhfeng) [inline components/camel-openapi-java/src/test/java/org/apache/camel/openapi/RestOpenApiReaderTest.java:257]: It looks like swagger v2 does not convert `Boolean` value.  see

https://github.com/swagger-api/swagger-core/blob/v1.6.11/modules/swagger-models/src/main/java/io/swagger/models/parameters/AbstractSerializableParameter.java#L173-L192

So there is a String `true` and `false` . @jamesnetherton is this a problem?

**Review by oscerd — APPROVED** [review]

File: components/camel-openapi-java/src/test/java/org/apache/camel/openapi/RestOpenApiReaderTest.java (hunk 1 above)
Comment (jamesnetherton) [reply components/camel-openapi-java/src/test/java/org/apache/camel/openapi/RestOpenApiReaderTest.java:257]: I guess it's not a regression as such, right? The `else` clause for the type check would have been doing fallback to `StringSchema` in past releases anyway.

Comment (zhfeng) [reply components/camel-openapi-java/src/test/java/org/apache/camel/openapi/RestOpenApiReaderTest.java:257]: Yeah, I think so.

**Review by jamesnetherton — APPROVED** [review]

**Pull request merged**